x/staking/keeper/historical_info.go

```diff
@@ -90,7 +90,7 @@ func (k Keeper) TrackHistoricalInfo(ctx sdk.Context) {
 	}
 
 	// Create HistoricalInfo struct
-	lastVals := k.GetLastValidators(ctx)
+	lastVals := k.GetBondedValidatorsByPower(ctx)
```

Comment:
so this doesn't check for max number of validators, only for bonded validators. I think we will need to create a new iterator function for this.

Reply:
What do you mean by this? Isn't that this line here:
https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/keeper/validator.go#L256

My intention is to figure out the active validator set. Will this function do that? If not, there surely is one in `x/staking`. I changed it to this function since I thought it gave me the same valset as before but more sorted

Reply:
cc: @cwgoes

Reply:
I believe that this function will fetch all bonded validators, but I'm not sure that this is the same sort as we need for Tendermint-compatibility, and I don't see how it matters since we're sorting it anyways? Also, this might not be the same as the "last validators", if the validator set has changed, no?

Reply:
yeah so this function will return the bonded validators, which can be a larger set than the active validator set

Reply:
Using `LastValidators`